cdap-ui/app/cdap/components/DataPrep/Directives/MergeColumns/index.js

```diff
@@ -0,0 +1,302 @@
+    if (nextProps.isOpen !== this.state.isOpen) {
+      this.setState({
+        isOpen: nextProps.isOpen
+      }, () => {
+        if (this.newColumnInputBox) {
+          this.newColumnInputBox.focus();
+        }
+      });
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.state.selectedDelimiter.substr(0, 6) === 'CUSTOM' && this.state.customDelimiter.length === 0 && this.customDelimiterRef) {
+      this.customDelimiterRef.focus();
+    }
+  }
+
+  preventPropagation(e) {
+    e.stopPropagation();
+    e.nativeEvent.stopImmediatePropagation();
+    e.preventDefault();
+  }
+
+  handleNewColumnInputChange(e) {
+    this.setState({newColumnInput: e.target.value});
+  }
+
+  handleDelimiterSelect(e) {
+    this.setState({selectedDelimiter: e.target.value});
+  }
+
+  handleCustomDelimiterChange(e) {
+    this.setState({customDelimiter: e.target.value});
+  }
+
+  handleKeyPress(e) {
+    if (e.nativeEvent.keyCode === 13) {
+      this.applyDirective();
+      this.preventPropagation(e);
+    }
+  }
+
+  columnAlreadyExists(colName) {
+    let headers = DataPrepStore.getState().dataprep.headers;
+    return headers.indexOf(colName) !== -1;
+  }
+
+  switchColumnOrder() {
+    if (this.state.newColumnInput === `${this.state.firstColumn}_${this.state.secondColumn}`) {
+      this.setState({
+        newColumnInput: `${this.state.secondColumn}_${this.state.firstColumn}`
+      });
+    }
+    this.setState({
+      firstColumn: this.state.secondColumn,
+      secondColumn: this.state.firstColumn
+    });
+  }
+
+  applyDirective() {
+    let newColumnInput = this.state.newColumnInput;
+    let delimiterSymbol = DELIMITER_MAP[this.state.selectedDelimiter];
+
+    if (this.state.selectedDelimiter.substr(0, 6) === 'CUSTOM') {
+      delimiterSymbol = this.state.customDelimiter;
+    }
+
+    let directive = `merge ${this.state.firstColumn} ${this.state.secondColumn} ${newColumnInput} ${delimiterSymbol}`;
+
+    execute([directive])
+      .subscribe(() => {
+        this.props.close();
+        this.props.onComplete();
+      }, (err) => {
+        console.log('error', err);
+
+        DataPrepStore.dispatch({
+          type: DataPrepActions.setError,
+          payload: {
+            message: err.message || err.response.message
+          }
+        });
+      });
+  }
+
+  renderCustomDelimiter() {
+    if (this.state.selectedDelimiter.substr(0, 6) !== 'CUSTOM') { return null; }
+
+    return (
+      <div>
+        <br />
+        <textarea
+          className="form-control"
+          value={this.state.customDelimiter}
+          onChange={this.handleCustomDelimiterChange}
+          ref={ref => this.customDelimiterRef = ref}
+          placeholder={T.translate(`${PREFIX}.customDelimiterPlaceholder`)}
+        />
+      </div>
+    );
+  }
+
+  renderDetail() {
+    if (!this.state.isOpen) { return null; }
+    let disabledCondition = this.state.newColumnInput.length === 0
+    || (this.state.selectedDelimiter.substr(0, 6) === 'CUSTOM'
```

Comment:
Any specific reason why we are not doing a whole word comparison? I am seeing this 
```
<option value="CUSTOMDELIMITER">
  {T.translate(`${PREFIX}.Delimiters.CUSTOMDELIMITER`)}
</option>
```
wouldn't `this.state.selectedDelimiter === 'CUSTOMDELIMITER'` easier? Also noticing the same check being performed in multiple places. Is it possible to extract this out as a function and reuse that everywhere?